app/com/linkedin/drelephant/tuning/hbt/FitnessManagerHBT.java

```diff
@@ -0,0 +1,246 @@
+
+    List<TuningJobExecutionParamSet> tuningJobExecutionParamSets = TuningJobExecutionParamSet.find.select("*")
+        .fetch(TuningJobExecutionParamSet.TABLE.jobExecution, "*")
+        .fetch(TuningJobExecutionParamSet.TABLE.jobSuggestedParamSet, "*")
+        .where()
+        .or(Expr.or(Expr.eq(TuningJobExecutionParamSet.TABLE.jobExecution + '.' + JobExecution.TABLE.executionState,
+            JobExecution.ExecutionState.SUCCEEDED),
+            Expr.eq(TuningJobExecutionParamSet.TABLE.jobExecution + '.' + JobExecution.TABLE.executionState,
+                JobExecution.ExecutionState.FAILED)),
+            Expr.eq(TuningJobExecutionParamSet.TABLE.jobExecution + '.' + JobExecution.TABLE.executionState,
+                JobExecution.ExecutionState.CANCELLED))
+        .isNull(TuningJobExecutionParamSet.TABLE.jobExecution + '.' + JobExecution.TABLE.resourceUsage)
+        .eq(TuningJobExecutionParamSet.TABLE.jobSuggestedParamSet + "." + JobSuggestedParamSet.TABLE.tuningAlgorithm
+            + "." + TuningAlgorithm.TABLE.optimizationAlgo, TuningAlgorithm.OptimizationAlgo.HBT.name())
+        .findList();
+
+    logger.debug("#completed executions whose metrics are not computed: " + tuningJobExecutionParamSets.size());
+
+    getCompletedExecution(tuningJobExecutionParamSets, completedJobExecutionParamSet);
+
+    return completedJobExecutionParamSet;
+  }
+
+  @Override
+  protected void calculateAndUpdateFitness(JobExecution jobExecution, List<AppResult> results,
+      TuningJobDefinition tuningJobDefinition, JobSuggestedParamSet jobSuggestedParamSet) {
+    logger.debug("calculateAndUpdateFitness");
+    Double totalResourceUsed = 0D;
+    Double totalInputBytesInBytes = 0D;
+    Double score=0D;
+    for (AppResult appResult : results) {
+      totalResourceUsed += appResult.resourceUsed;
+      totalInputBytesInBytes += getTotalInputBytes(appResult);
+      score += appResult.score;
+    }
+
+    Long totalRunTime = Utils.getTotalRuntime(results);
+    Long totalDelay = Utils.getTotalWaittime(results);
+    Long totalExecutionTime = totalRunTime - totalDelay;
+
+    if (totalExecutionTime != 0) {
+      jobExecution.score=score;
+      updateJobExecution(jobExecution, totalResourceUsed, totalInputBytesInBytes, totalExecutionTime);
+    }
+
+    if (tuningJobDefinition.averageResourceUsage == null && totalExecutionTime != 0) {
+      updateTuningJobDefinition(tuningJobDefinition, jobExecution);
+    }
+
+    //Compute fitness
+    computeFitness(jobSuggestedParamSet, jobExecution, tuningJobDefinition, results);
+  }
+
+  protected void computeFitness(JobSuggestedParamSet jobSuggestedParamSet, JobExecution jobExecution,
+      TuningJobDefinition tuningJobDefinition, List<AppResult> results) {
+    if (!jobSuggestedParamSet.paramSetState.equals(JobSuggestedParamSet.ParamSetStatus.FITNESS_COMPUTED)
+        || !jobSuggestedParamSet.paramSetState.equals(JobSuggestedParamSet.ParamSetStatus.DISCARDED)) {
```

Comment:
Not sure, but looks like here AND condition should be used.

Reply:
Fixed in https://github.com/linkedin/dr-elephant/pull/450